src/components/button/types.d.ts

```diff
@@ -0,0 +1,5 @@
+export type Props = {
+  name: string;
+  text: string;
+  type?: string;
```

Comment:
Полагаю, что тип у кнопки может принимать ограниченое количество значений, и если положить в него любую строку, то это будет ошибка. Предлагаю в таком случае воспользоваться discriminated union
```suggestion
  type?: 'button' | 'submit' и тд;
```